alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/UndeployTask.java

```diff
@@ -68,40 +69,49 @@ public void run() {
             }
         }
 
-        log.debug("Undeploying deployment Id:" + paasId);
-        boolean done = false;
-        String taskUrl = null;
-        String status;
+        log.debug("Undeploying with purge deployment Id:" + paasId);
         try {
-            taskUrl = restClient.undeploy(deploymentUrl, false);
+            taskUrl = restClient.undeploy(deploymentUrl, true);
             if (taskUrl == null) {
                 // Assumes already undeployed
                 orchestrator.changeStatus(paasId, DeploymentStatus.UNDEPLOYED);
                 done = true;
             }
         }
-        catch (YorcRestException jre) {
+        catch(YorcRestException jre){
+            // If 400 code (bad request)  is returned, we retry requesting purge during at most 5 minutes
+            if (jre.getHttpStatusCode() == 400) {
```

Comment:
Why do we implement a retry here?

Reply:
It's legacy. Don't know why it has been done